src/nvim/menu.c

```diff
@@ -1173,7 +1173,7 @@ static char_u *menu_text(char_u *str, int *mnemonic, char_u **actext)
   if (p != NULL) {
     if (actext != NULL)
       *actext = vim_strsave(p + 1);
-    text = vim_strnsave(str, (int)(p - str));
+    text = vim_strnsave(str, (size_t)(p - str));
```

Comment:
In general, before casting, you need to make sure that it will give a valid result. In this case, adding an `assert(p >= str)` should work.